### ISSUE TITLE
OpenCL error -61 - cannot allocate big buffer for DAG (Vega 64 8GB)

### ISSUE THREAD

**Issue by Wacholek** [Claymore-Dual/Claymore-Dual-Miner#31]
I have two rigs. One with 2 Vega 64 8GB and one with 3 Vega 64 8GB. 
Up to yesterday there was no problem. Both on Windows 10. Different builds ( update disabled, drivers update disabled). On one rig I had to restart the miner but after the restart that message showed up:

GPU0 - not enough GPU memory to place DAG, you cannot mine this coin with this GPU
GPU2 - not enough GPU memory to place DAG, you cannot mine this coin with this GPU
GPU1 - not enough GPU memory to place DAG, you cannot mine this coin with this GPU
GPU0 - OpenCL error -61 - cannot allocate big buffer for DAG. Check readme.txt for possible solutions.
GPU2 - OpenCL error -61 - cannot allocate big buffer for DAG. Check readme.txt for possible solutions.
GPU1 - OpenCL error -61 - cannot allocate big buffer for DAG. Check readme.txt for possible solutions.
GPU2, OpenCL error -38 - cannot write buffer for DAG
GPU1, OpenCL error -38 - cannot write buffer for DAG
GPU0, OpenCL error -38 - cannot write buffer for DAG'

In the same time on the second rig Claymore's Miner was mining without a problem. So I left that one and started to look for the problem on the first one.
After an hour of try and errors I moved to another miner (SRBminer) and there was no problem with mining.
On the second rig Claymore's Miner was still mining fine. So for the test purposes I restart this one to and it throw me the same error as on the first rig.

On both rigs I have 8GB or RAM and 64GB of swap file on SSD drive.
Claymore's Miner is started with administration rights and with:
GPU_FORCE_64BIT_PTR 0
GPU_MAX_HEAP_SIZE 100
GPU_USE_SYNC_OBJECTS 1
GPU_MAX_ALLOC_PERCENT 100
GPU_SINGLE_ALLOC_PERCENT 100

On both rigs SRBminer and TeamRedMiner is running just fine.
I mine ETC. SRBminer shows that it is epoch 377 ( DAG size 4039MB) while Claymore's Miner DAG epoch #377(3.95GB).
ETH is mining ok.

**Comment by Tr0nicje** [comment]
I'm having the exact same issue mining ETC with RX 580 8GB cards now

**Comment by 321mindminer** [comment]
Hi I also started to facing with problem since yesterday, 9.10 20:40. since that moment both my rigs which were mining ETC reports DAG size problem. All cards are rx580 8gb, so not the hardware is a problem. I’m using HiveOS. Of course any tricks like eres 0 doesn’t work. Maybe New epoch triggered some issue with claymore or ETC team did sth wrong? On ETH rigs are working OK. Problem should be investigated by claymore team I think.

**Comment by bendusz** [comment]
I have the exact same issue on my 8x Vega56 rig where I was mining ETC. When I tested on only one card, it is working fine, but with the 8 card rig, it just throws an error for each card that it can not load DAG.

**Comment by MrSevy-zero** [comment]
I wanted to switch to etc mining ( from eth ) and I'm having the same problem with 8GB rx480 rx470 :(

**Comment by bundela153** [comment]
Unfortunately the issue will persist for ETH too when dag size reaches 3.95 GB. The issue is with claymore only. Nicehash also post about the same issue.

**Comment by 321mindminer** [comment]
> Unfortunately the issue will persist for ETH too when dag size reaches 3.95 GB. The issue is with claymore only. Nicehash also post about the same issue.

It's very possible and you're right, problem is with claymore. I regret that I had to change the miner because I had well optimized my rigs. But, phoenixminer works with the same speed, even 1-2MH faster and takes only a few Watts more :)
I think that claymore will do sth with that soon, because he will lose profits.

**Comment by Tr0nicje** [comment]
I've got Claymore miner working again on my 9x RX580 8GB rig to mine ETC!
The AMD drivers installed were 20.4.2 and I was having the DAG size problems.
Now updated to latest 20.9.2 and after a restart I've been able to mine ETC stable for the past 24hours. 
![afbeelding](https://user-images.githubusercontent.com/72630035/96386265-d3710500-1199-11eb-82fe-9c7a3f8f1dfd.png)

**Comment by BFCE** [comment]
Same is happening to me. 18.6.1 drivers, RX VEGA 56 and RX 480 8GB

**Comment by BlanketLord** [comment]
updating to 20.9.2 did temporarily fix the issue but as of today it has resurfaced 
adding -eres 0 seems to have temporarily fixed it 
6 - RX 570 8GB

**Comment by mbunal** [comment]
> updating to 20.9.2 did temporarily fix the issue but as of today it has resurfaced
> adding -eres 0 seems to have temporarily fixed it
> 6 - RX 570 8GB

-eres 0 doesnt work in linux. Works only in windows.

**Comment by ejtttje** [comment]
> -eres 0 doesnt work in linux. Works only in windows.

Works for me in linux.  Also got it to start mining for me again.  (-eres 1 also works, but 2 does not.)

I have a feeling that's only buying a few more epochs...?

I think 4GB cards are down for the count though, time to put any 4GB cards out to pasture..

**Comment by 321mindminer** [comment]
Yes, claymore is working again now, but it's futureless. It seems that project is no longer being developed. I swithed to phoenixminer and this miner works better, more efficiently and is still mantained. I don't know which OS you're using, but a few days ago HiveOS team removed claymore-dual miner from the list of available miners. It seems that this is the end of claymore.
